capstone/src/main/java/com/google/sps/data/ReadData.java

```diff
@@ -0,0 +1,71 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReadData {
+  final String DELIMITER = ",";
+
+  public void readCSV() {
+    BufferedReader reader = null;
+    FileReader file = null;
+    DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+    try {
```

Comment:
This is a large try catch loop. You usually want to keep try catches as small as possible. And I think it's because you're try catching multiple different errors. Heres what you can do to simplify this.

Firstly, for most of the exceptions you're catching, you're simply printing the stack trace which is provided to you automatically. The only thing that might need a try catch is the reader because there is iterative work you want to do but also you want to ensure that you're closing the reader once all the work is done.

```
file = new FileReader("assets/surveyresponse.csv");
try {
    reader = new BufferedReader(file);
    ...
} finally {
    reader.close();
}
```

Reply:
I'll add that change

Reply:
I get errors for IO exceptions when using the file and the input streams